backend/src/main/java/com/redhat/cloud/notifications/events/EventConsumer.java

```diff
@@ -52,20 +53,23 @@
     private Counter rejectedCounter;
     private Counter processingErrorCounter;
     private Counter duplicateCounter;
-    private Counter consumedCounter;
 
     @PostConstruct
     public void init() {
         rejectedCounter = registry.counter(REJECTED_COUNTER_NAME);
         processingErrorCounter = registry.counter(PROCESSING_ERROR_COUNTER_NAME);
         duplicateCounter = registry.counter(DUPLICATE_COUNTER_NAME);
-        consumedCounter = registry.counter(CONSUMED_COUNTER_NAME);
     }
 
     @Incoming(INGRESS_CHANNEL)
     @Acknowledgment(PRE_PROCESSING)
     public Uni<Void> processAsync(Message<String> message) {
+        // This timer will have dynamic tag values based on the action parsed from the received message.
+        Timer.Sample consumedTimer = Timer.start(registry);
         String payload = message.getPayload();
+        // The two following variables have to be final or effectively final. That why their type is String[] instead of String.
+        String[] bundleName = new String[1];
+        String[] appName = new String[1];
```

Comment:
Looks like these are pointers to a String

```suggestion
        String *bundleName;
        String *appName;
```

only kidding :grin:

Reply:
That's the idea 😄